### DISCUSSION TITLE
Increase volume

### DISCUSSION THREAD

**Discussion by ther3zz** [tetele/onju-voice-satellite#64] [Q&A]
It seems that the volume is lower using the onju board vs using the normal google board. I've already set the speaker volume to the maximum and even still, it seems much lower.

Is there any way to further increase this?

**Comment by naorunaoru** [comment]
One of the differences between Nest and Onju is that the Nest allows for hardware digital sound processing. 

There are two ways to increase perceptible volume: the straightforward one would be increasing the signal amplitude coming to the speaker, which is limited by the power budget of the amplifier and by the speaker's maximum rated input, both of which are rather low. The other way is to use a compressor or a limiter on the output chain. 

Google's implementation is a bit more involved: my assumption is that they use some combination of an equalizer and a multiband compressor to correct the speaker's weak frequency response, increase perceptible volume and also provide some form of loudness equalization by using an equal loudness curve. 

The question is: can we implement the same on ESP32 without a dedicated DSP chip? 

EDIT: and the answer is a resounding "maybe": ESP-DSP provides at least some of the necessary low-level abstractions like FFT and inverse FFT, but the performance implications of running a heavy audio pipeline are unclear.

**Reply by ther3zz** [reply]
maybe is certainly better than a "no"!

**Reply by naorunaoru** [reply]
I'd suggest pulling @gnumpi into this discussion if they don't mind, because esphome_audio would be the best place to implement these features as it is already a very good wrapper around ESP-ADF.
